Feature/add mesh guide (#4081)
fishtown-analytics/docs.getdbt.com: website/docs/guides/best-practices/how-we-mesh/mesh-1-intro.md

---
title: "Intro to dbt Mesh"
description: Getting started with dbt Mesh patterns
hoverSnippet: Learn how to get started with dbt Mesh
---

## What is dbt Mesh?

> Historically, building data teams has involved two extremes, building a centralized team or using embedded analysts. More recently, hub-and-spoke models have become popular as a way to balance the tradeoffs: using a centralized platform team _and_ embedded analysts, allowing embeds to develop domain expertise while the central team focuses on building a strong operational foundation. A major difficultly of this model though is managing the compplexity of dependencies, goverance, and workflows between all groups — creating friction in monorepos or complexity and silos in multi-repos. Ideally, you want to teams to be able to work independently, but also be able to collaborate; sharing data, code, and best practices. dbt Mesh provides the tooling for teams to finally achieve this.

Reviewer: ```suggestion
Historically, building data teams has involved two extremes, building a centralized team or using embedded analysts. More recently, hub-and-spoke models have become popular as a way to balance the tradeoffs: using a centralized platform team _and_ embedded analysts allows embeds to develop domain expertise while the central team focuses on building a strong operational foundation. A major difficultly of this model is managing the complexity of dependencies, governance, and workflows between all groups — attempting to manage this can often create friction in monorepos or complexity and silos in multiple repos. Ideally, teams should be able to work independently and also be able to collaborate; sharing data, code, and best practices. dbt Mesh provides the tooling for teams to finally achieve this.
```

> dbt Mesh is not a product, but a pattern, enabled a convergence of several features in dbt Cloud. It’s inspired by dbt’s best practices and ideas from [data mesh](https://en.wikipedia.org/wiki/Data_mesh). These features include:

Reviewer: ```suggestion
dbt Mesh is not a product, but a pattern, enabled by a convergence of several features in dbt Cloud. It’s inspired by dbt’s best practices and ideas from [data mesh](https://en.wikipedia.org/wiki/Data_mesh). These features include:
```

> - **Cross-project references** - this is the core feature that enables a mesh structure. `ref`s now work across projects in dbt Cloud-enabled projects on Enterprise plans.

Reviewer: may want to avoid the word core here -- maybe foundational?

> - **Governance** - dbt Cloud’s new governance features allow you to manage access and permissions across projects.

Reviewer: ```suggestion
- **Governance** - dbt's new governance features allow you to manage access and permissions across projects.
```

- **Contracts** - data contracts set strict expectations on the shape of the data to ensure data changes upstream of dbt or within a project's logic don't break downstream consumers.

## Who is dbt Mesh for?

dbt Mesh is not for every organization! If you're just starting your dbt journey, don't worry about building a dbt Mesh right away, it increases some meta-complexity around managing your projects that could distract from building initial value in dbt. However, if you're already using dbt and your project has started to experience any of the following, you're likely ready to start exploring a dbt Mesh:

- **The number of models** in your project is degrading performance and slowing down development.
- Teams have developed **separate workflows** and need to decouple development.
- **Security and governance** requirements are increasing and would benefit from increased isolation.

dbt Cloud is designed to coordinate the features above and simplify the meta-complexities (such as scoped CI and multi-project lineage) to solve for these problems.

> ## Learning goals

Reviewer: emoji me cap'n

---
title: "Intro to dbt Mesh"
description: Getting started with dbt Mesh patterns
hoverSnippet: Learn how to get started with dbt Mesh
---

## What is dbt Mesh?

Organizations of all sizes rely upon dbt to manage their data transformations, from small startups to large enterprises. At scale, it can be challenging to coordinate all the organizational and technical requirements demanded by your stakeholders within the scope of a single dbt project. To date, there also hasn't been a first-class way to effectively manage the dependencies, governance, and workflows between multiple dbt projects.

Regardless of your organization's size and complexity, dbt should empower data teams to work independently and collaboratively; sharing data, code, and best practices without sacrificing security or autonomy. dbt Mesh provides the tooling for teams to finally achieve this.

dbt Mesh is not a single product: it is a pattern enabled by a convergence of several features in dbt:

> - **[Cross-project references](/docs/collaborate/govern/project-dependencies#usage)** - this is the foundational feature that enables the multi-project deployments. `{{ ref() }}`s now work across dbt Cloud projects on Enterprise plans.

Reviewer 2: ```suggestion
- **[Cross-project references](/docs/collaborate/govern/project-dependencies#how-to-use-ref)** - this is the foundational feature that enables the multi-project deployments. `{{ ref() }}`s now work across dbt Cloud projects on Enterprise plans.
```

- **[dbt Explorer](/docs/collaborate/explore-projects)** - dbt Cloud's metadata-powered documentation platform, complete with full, cross-project lineage.
- **Governance** - dbt's new governance features allow you to manage access to your dbt models both within and across projects.

>   - **[Groups](/docs/build/groups)** - groups allow you to assign models to subsets of models within a project.

Reviewer 3: I found this a little confusing. Is this also correct? If not, can you reword to clarify? Also updating the link. 
```suggestion
  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to assign models to subsets within a project.
```

>   - **[Access](/docs/collaborate/govern/model-access)** - access configs allow you to control who can reference models.

Reviewer 3: I think a more direct link to the subsection might be helpful here:
```suggestion
  - **[Access](/docs/collaborate/govern/model-access#access-modifiers)** - access configs allow you to control who can reference models.
```

> - **[Model Versions](/docs/collaborate/govern/model-versions)** - When coordinating across projects and teams, we recommend treating your data models as stable APIs. Model versioning is the mechanism to allow graceful adoption and deprecation of models as they evolve.

Reviewer 3: Just making capitalization consistent:
```suggestion
- **[Model Versions](/docs/collaborate/govern/model-versions)** - when coordinating across projects and teams, we recommend treating your data models as stable APIs. Model versioning is the mechanism to allow graceful adoption and deprecation of models as they evolve.
```

> - **[Model Contracts](/docs/collaborate/govern/model-contracts)** - data contracts set explicit expectations on the shape of the data to ensure data changes upstream of dbt or within a project's logic don't break downstream comsumers' data products.

Reviewer 3: ```suggestion
- **[Model Contracts](/docs/collaborate/govern/model-contracts)** - data contracts set explicit expectations on the shape of the data to ensure data changes upstream of dbt or within a project's logic don't break downstream consumers' data products.
```

## Who is dbt Mesh for?

> The multi-project architecture is for organizations with mature, complex transformation workflows in dbt who want to increase the flexibilty and performance of their dbt projects. If you're already using dbt and your project has started to experience any of the following, you're likely ready to start exploring this paradigm:

Reviewer 3: Just a tiny tweak and typo fix:
```suggestion
The multi-project architecture helps organizations with mature, complex transformation workflows in dbt increase the flexibility and performance of their dbt projects. If you're already using dbt and your project has started to experience any of the following, you're likely ready to start exploring this paradigm:
```

> - **The number of models** in your project is degrading performance and slowing down development.

Reviewer 3: Suggest highlighting the meaty bits only:
```suggestion
- The **number of models** in your project is degrading performance and slowing down development.
```
